### DISCUSSION TITLE
"MSI2AAE.tmp is not a valid Win32 application" during install

### DISCUSSION THREAD

**Discussion by MartinDWhite** [wixtoolset/issues#7271] [Questions]
We built an .msi install file with WixSharp in VS 2022. It has worked on many machines. It has started to fail on some machines with an error dialog. The title of the dialog is "RunDLL" and the contents are 
"There was a problem starting C:\Windows\Installer\MSI2AAE.tmp 
C:\Windows\Installer\MSI2AAE.tmp is not a valid Win32 application."

The error is before the user is asked any of the questions during the install. 
This error dialog comes up before the install progress bar has started to move at all. 

This does not occur on all machines, and does not happen on any of our test machines.  
It does happen on brand a new Windows 11 machine for one of our clients.
It does not happen on all windows 11 machines. 

Here is a screen shot from the user showing the error dialog. 
![error_screen_shot](https://user-images.githubusercontent.com/14093566/224738958-08c0f23e-2b81-4ac8-9218-421fda490e7f.png)

A bunch of internet searching and all that I found was this post which appears to be the same error with a different MSI*.tmp file name. https://stackoverflow.com/questions/75673892/rundll-error-message-msib4de-tmp-is-not-a-valid-win32-application-when-instal 

Any thoughts on something we can do to fix this?

**Comment by robmen** [comment]
The StackOverflow link suggests the user needs to apply their Windows Updates. Maybe that will work for you as well.

**Comment by MartinDWhite** [comment]
I was able to get connected to the computer in question, where the install failed, and the install worked with the same install file and no other changes. There were no logs of windows updates being run on the machine between the time the install failed and worked. This is one of many installs with this same install file and all the others have worked without issue. For lack of any other way to move forward, I am going to assume something was broken on that particular machine.
